src/kaltura-live-plugin.tsx

```diff
@@ -282,7 +284,7 @@ export class KalturaLivePlugin implements OnMediaUnload, OnPluginDestroy {
         this._initTimeout();
         switch (type) {
             case OfflineTypes.NoLongerLive:
-                if (this._attachMedia) {
+                if (!this._player.isDvr() || this._attachMedia) {
```

Comment:
seems like you assume `_attachMedia` is true in dvr only, right? 
the question is dont you need to handle primary to secondary switching for non dvr as well?

Reply:
`_attachMedia` happens only if media ended (`ended` event dispatched) that happens only for DVR entries
switching to primary\secondary happens without `_attachMedia`, it happens by `error` => `reload media`